common:
    lid_topic:  "/os1_points"
    imu_topic:  "/imu/data_raw"
    camera_topic: "/usb_cam/image_raw"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible

    rootDir: "/home/yixin-f/fast-lio2/src/data/"
    savePCD: true                         # https://github.com/TixiaoShan/LIO-SAM/issues/3
    savePCDDirectory: "PCDs/"        
    saveSCD: true 
    saveSCDDirectory: "SCDs/"
    saveLOG: true
    saveLOGDirectory: "LOG/"
    map_save_en: true  # global map
    pcd_save_interval: 1                 # how many LiDAR frames saved in each pcd
    
preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR
    livox_type: 1                # 1 for LIVOX_CUS
    scan_line:  64
    blind: 4
    feature_enabled: 0  # bool
    point_filter_num: 1 # bool
    scan_rate: 10 

camera:
    camera_en: false
    camera_external: [ 0.000000, -1.000000, 0.000000, 0.000000,
                       0.000000, 0.000000, -1.000000, 0.300000,
                       1.000000, 0.000000, 0.000000, 0.000000,
                       0.000000, 0.000000, 0.000000, 1.000000]        #相机2雷达外参
    camera_internal: [516.235840, 0.000000, 641.125888, 0.000000,
                      0.000000, 533.185608, 338.725370, 0.000000,
                      0.000000, 0.000000, 1.000000, 0.000000]         #相机内参

mapping:
    acc_cov: 0.009939570888238808e-03
    gyr_cov: 0.005636343949698187e-03
    b_acc_cov: 0.64356659353532566e-03
    b_gyr_cov: 0.35640318696367613e-03
    fov_degree:    100
    det_range:     260.0
    cube_len: 200   # range
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic
    mappingSurfLeafSize: 0.2
    keyframeAddingDistThreshold: 1.0
    keyframeAddingAngleThreshold: 0.2
    extrinsic_T: [1.77, -0.00, -0.05]
    extrinsic_R: [ -1, 0, 0,
                   0, -1, 0,
                   0, 0, 1]

ikdtree:
    max_iteration: 4
    recontructKdTree: true  # if path is too large, the rviz will crash, but default is false
    filter_size_map_min: 0.4

loop:
    loopClosureEnableFlag: true
    loopClosureFrequency: 1.0
    historyKeyframeSearchRadius: 10.0
    historyKeyframeSearchTimeDiff: 30.0
    historyKeyframeSearchNum: 1
    historyKeyframeFitnessScore: 0.3

segment:
    ground_en: false
    tollerance_en: false  # // TODO: error !
    sensor_height: 1.5
    z_tollerance: 2.0
    rotation_tollerance: 0.2

publish:
    path_en:  true
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: false       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

visualization:
    globalMapVisualizationSearchRadius: 10.0
    globalMapVisualizationPoseDensity: 5
    globalMapVisualizationLeafSize: 0.5
    visulize_IkdtreeMap: true
